src/pages/default/connections/AddConnectionScreen.tsx

```diff
@@ -0,0 +1,323 @@
+import React, { useEffect, useState } from "react";
+import { SafeAreaView, Text, View, StyleSheet, Alert } from "react-native";
+import {
+  BarCodeScanner,
+  type BarCodeScannerResult,
+} from "expo-barcode-scanner";
+import queryString from "query-string";
+import { v4 as uuid } from "uuid";
+import { base58btc } from "multiformats/bases/base58";
+import { base64url } from "multiformats/bases/base64";
+import { hkdf } from "@noble/hashes/hkdf";
+import { sha256 } from "@noble/hashes/sha256";
+import { xchacha20_poly1305 } from "@noble/ciphers/chacha";
+import { edwardsToMontgomeryPub, x25519 } from "@noble/curves/ed25519";
+import { createJsonRpcRequest } from "@tbd54566975/web5-agent";
+import { FlexLayouts, Layouts } from "@/theme/layouts";
+import type { AppNavigatorProps } from "@/types/navigation";
+
+type Props = AppNavigatorProps<"AddConnectionScreen">;
+const AddConnectionScreen = ({ navigation }: Props) => {
+  const [hasPermission, setHasPermission] = useState<boolean | null>(null);
+  const [isScanned, setIsScanned] = useState(false);
+
+  useEffect(() => {
+    const getBarCodeScannerPermissions = async () => {
+      const { status } = await BarCodeScanner.requestPermissionsAsync();
+      setHasPermission(status === "granted");
+    };
+
+    void getBarCodeScannerPermissions();
+  }, []);
+
+  const onQRCodeScanned = ({ data }: BarCodeScannerResult) => {
+    setIsScanned(true);
+
+    try {
+      // pull out the query params from the QR code
+      const qrParams = extractQueryParams(data);
+
+      initConnect(qrParams.dwaDID, qrParams.connectNonce, qrParams.serverURL);
+    } catch (e: any) {
+      alert(e.message);
+      navigation.goBack();
```

Comment:
Something to think about for later: 
I've often had QR code scanning fail on my first attempt (eg: blurry camera and it couldn't parse the URL properly) on my iPhone 13 Mini. It _could_ be nice to keep the user on this screen to try again in situations like this.

I think for now, this is totally fine. But we might want to gather some feedback from others as we alpha/beta test the app to see if this is a common problem or not.

Reply:
Updated with your callout and added a todo